tests/unit/onyxTest.js

```diff
@@ -488,23 +488,50 @@ describe('Onyx', () => {
                 value: 'three',
             },
         };
-        Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_CONNECT_COLLECTION, collectionData);
+
+        Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_CONNECT_COLLECTION, initialCollectionData);
         return waitForPromisesToResolve()
             .then(() => {
-                // WHEN we call Onyx.connect with the waitForCollectionCallback = true param
-                connectionID = Onyx.connect({
+                // WHEN we connect to that collection with waitForCollectionCallback = true
+                Onyx.connect({
                     key: ONYX_KEYS.COLLECTION.TEST_CONNECT_COLLECTION,
                     waitForCollectionCallback: true,
                     callback: mockCallback,
                 });
                 return waitForPromisesToResolve();
             })
             .then(() => {
-                // THEN the callback should be triggered only once
+                // THEN we expect the callback to be called only once and the initial stored value to be initialCollectionData
                 expect(mockCallback.mock.calls.length).toBe(1);
+                expect(mockCallback.mock.calls[0][0]).toEqual(initialCollectionData);
```

Comment:
Should we have a version of this test where you connect to a collection that has not been initialized? what would you have in `mockCallback.mock.calls[0][0]` in such case? In the test below this happens... but we don't check the value in `mockCallback.mock.calls[0][0]`

Reply:
The 2nd test does this and I think the expected value would be the object below. That's because we set those keys in the 1st test and we clear them (set the values to null, but the keys still exist) [after each test](https://github.com/Expensify/react-native-onyx/blob/b348a7d7b448c02d6994f31feccb208db4049070/tests/unit/onyxTest.js#L26). 

```
{"test_connect_collection_1": null, "test_connect_collection_2": null, "test_connect_collection_3": null}
```

We can have it be null if we use a new collection key though

Reply:
I updated the tests with a new collection key, so we can check for a null value easily. Let me know if you had something different in mind!

Reply:
yes, that sounds good... I think it is bad to reuse keys between different tests, it may cause some bad side effect / dependency between them

Reply:
Cool! Updated!